Add raw parameter on AdapterInterface and let OpenSpout exporter use it

mhvis opened the pull request:
This prevents calling normalize() and render() on the column during export with the ExcelOpenSpoutExporter.

This allows the exporter to decide how PHP values are represented in the output file.

mhvis committed:
Add raw parameter on AdapterInterface and let OpenSpout exporter use it

This prevents calling normalize() and render() on the column during
export with the ExcelOpenSpoutExporter.

This allows the exporter to decide how PHP values are represented in the
output file.

curry684 commented:
@NMe84 also relevant for you I believe.

curry684 commented:
Can't merge btw because of conflicts with other PR.

curry684 commented:
Happy to add to 0.10 if fixed in time, released RC today for early testing: https://github.com/omines/datatables-bundle/releases/tag/0.10.0-rc.1

mhvis committed:
Do not cache OptionsResolvers in `AbstractColumn::$resolversByClass`

This fixes a tricky bug. When you initialize a BoolColumn for the second
time, the previously configured OptionsResolver is used. In this
resolver, the default `rightExpr` option value is a callable where $this
refers to the BoolColumn that was created during the first
initialization. This might be a different BoolColumn instance. The call
to $this->getTrueValue() inside the `rightExpr` callable will then
return the wrong value.

I'm not sure if this resolver 'cache' was added after profiling or as a
precautionary measure. If it was just a precaution, I would remove it as
in this commit, as performance optimizations should always only be added
as a result of profiling.

If this was truly done to fix a performance issue, then a different fix
is necessary.

mhvis commented:
Thanks for merging the other PRs!

I had to add a new change after resolving the conflicts, to fix a test case. *Please read the commit message*: 703a5c21b06a2fba7459be0dd97638aa15ec852c

curry684 commented:
The cache should be useless so fine like this.

curry684 commented:
https://github.com/omines/datatables-bundle/releases/tag/0.10.0-rc.2

mhvis commented:
Thanks!! Will test the release candidate as soon as possible
